src/main/java/randoop/output/FailingTestFilter.java

```diff
@@ -0,0 +1,277 @@
+      }
+      javaCodeLines[lineNumber - 1] = "// flaky: " + javaCodeLines[lineNumber - 1];
+
+      failureCount++;
+    }
+
+    //XXX have this method return the array and redo writeClass so that it writes from array (?)
+    return UtilMDE.join(javaCodeLines, Globals.lineSep);
+  }
+
+  /**
+   * Read lines of the JUnit output using the iterator until finding a match for the pattern, and
+   * then returns a pair containing the line and the text matching the first group of the pattern.
+   * Assumes that there is a match, and that the pattern has at least one group.
+   *
+   * @param lineIterator the iterator for reading from the JUnit output
+   * @param pattern the pattern for a regex with at least one group
+   * @return the pair containing the line and the text matching the first group
+   * @throws BugInRandoopException if the iterator has no more lines, but the pattern hasn't been
+   *     matched
+   */
+  private Pair<String, String> readUntilMatch(Iterator<String> lineIterator, Pattern pattern) {
+    while (lineIterator.hasNext()) {
+      String line = lineIterator.next();
+      Matcher matcher = pattern.matcher(line);
+      if (matcher.matches()) {
+        return Pair.of(line, matcher.group(1));
+      }
+    }
+    throw new BugInRandoopException("Error: JUnit output doesn't contain: " + pattern.pattern());
+  }
+
+  /**
+   * Compiles the Java files in the list of files and writes the resulting class files to the
+   * directory.
+   *
+   * @param packageName the package name for the test class
+   * @param classname the name of the test class
+   * @param classSource the text of the test class
+   * @param destinationDir the directory for class file output
+   * @throw BugInRandoopException if the file does not compile
+   */
+  private void compileTestClass(
+      String packageName, String classname, String classSource, Path destinationDir) {
+    // TODO: The use of FileCompiler is temporary. Should be replaced by use of SequenceCompiler,
+    // which will compile from source, once it is able to write the class file to disk.
+    List<File> sourceFiles = new ArrayList<>();
+    sourceFiles.add(javaFileWriter.writeClassCode(packageName, classname, classSource));
+    FileCompiler fileCompiler = new FileCompiler();
+    try {
+      fileCompiler.compile(sourceFiles, destinationDir);
+    } catch (FileCompilerException e) {
+      throw new BugInRandoopException("Compilation error during flaky-test filtering", e);
+    }
+  }
+
+  /**
+   * Creates a temporary directory by concatenating the class name and a pass count to form the
+   * directory name.
+   *
+   * @param classname the class name
+   * @param pass the pass count
+   * @return the {@code Path} for the directory created
+   */
+  private Path createWorkingDirectory(String classname, int pass) {
+    try {
+      Path workingDirectory = Files.createTempDirectory("check" + classname + pass);
+      //workingDirectory.toFile().deleteOnExit();
```

Comment:
Why is this commented out?  Should it be uncommented before the branch is merged into master?